app/src/main/java/org/jboss/hal/client/configuration/subsystem/ejb/EjbPresenter.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.hal.client.configuration.subsystem.ejb;
+
+import java.util.List;
+import javax.inject.Inject;
+
+import com.google.web.bindery.event.shared.EventBus;
+import com.gwtplatform.mvp.client.annotations.NameToken;
+import com.gwtplatform.mvp.client.annotations.ProxyCodeSplit;
+import com.gwtplatform.mvp.client.proxy.ProxyPlace;
+import org.jboss.hal.core.finder.Finder;
+import org.jboss.hal.core.finder.FinderPath;
+import org.jboss.hal.core.finder.FinderPathFactory;
+import org.jboss.hal.core.mbui.MbuiPresenter;
+import org.jboss.hal.core.mbui.MbuiView;
+import org.jboss.hal.core.mvp.HasVerticalNavigation;
+import org.jboss.hal.dmr.ModelDescriptionConstants;
+import org.jboss.hal.dmr.ModelNode;
+import org.jboss.hal.dmr.dispatch.Dispatcher;
+import org.jboss.hal.dmr.model.NamedNode;
+import org.jboss.hal.dmr.model.Operation;
+import org.jboss.hal.meta.StatementContext;
+import org.jboss.hal.meta.token.NameTokens;
+import org.jboss.hal.spi.Requires;
+
+import static org.jboss.hal.client.configuration.subsystem.ejb.AddressTemplates.*;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.READ_RESOURCE_OPERATION;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.RECURSIVE;
+import static org.jboss.hal.dmr.ModelDescriptionConstants.SERVICE;
+import static org.jboss.hal.dmr.ModelNodeHelper.asNamedNodes;
+import static org.jboss.hal.dmr.ModelNodeHelper.failSafePropertyList;
+
+/**
+ * @author Claudio Miranda
+ */
+public class EjbPresenter extends MbuiPresenter<EjbPresenter.MyView, EjbPresenter.MyProxy> {
+
+    // @formatter:off
+    @ProxyCodeSplit
+    @NameToken(NameTokens.EJB3)
+    @Requires({EJB_SUBSYSTEM_ADDRESS, THREAD_POOL_ADDRESS, REMOTING_PROFILE_ADDRESS, BEAN_POOL_ADDRESS, CACHE_ADDRESS, 
```

Comment:
Pretty much resources in here. How's the performance when first loading the EJB application? 

The r-r-d op uses `recursive-depth=2` by default. So a simple `@Requires(EJB_SUBSYSTEM_ADDRESS)` should work?